### PR TITLE
Add max swappable amount on balance badge tap

### PR DESCRIPTION
**Pull request by walmat**
Fixes APP-1557

## What changed (plus any additional context for devs)
Adds balance badge press setting max swappable amount for asset if user has balance.

Note I also added the functionality to output balance badge.

## Screen recordings / screenshots

https://github.com/user-attachments/assets/d123c20e-914d-496d-87ab-3729aa14e1e5

### CODE REVIEW

**Review by greg-schrammel — APPROVED** [review]
nice

**Review by jinchung — CHANGES_REQUESTED** [review]
1. I think having this functionality on the output badge will not be that useful @christianbaroni  if you have thoughts here
2. This adds duplicate code, can we have the `onSetMaxBalance` worklet pulled out separately and shared across the `SwapSlider` and `SwapInputAsset`? Assuming we don't do the output asset, this simplifies it even more. This is for maintenance considerations going forward, if we need to update this logic, it's easier to just update in one place than in two or three.

File: src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts [hunk 1]
```diff
@@ -72,6 +83,7 @@ export function useSwapInputsController({
   quote,
   sliderXPosition,
   slippage,
+  outputQuotesAreDisabled,
```
Comment (jinchung) [inline src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts:86]: this param no longer needed and the corresponding changes in `swap-provider` also not needed

**Review by jinchung — APPROVED** [review]
🌮

**Pull request merged**